core/templates/pages/library-page/library-page.component.ts

```diff
@@ -498,56 +499,59 @@ export class LibraryPageComponent {
   }
 
   moveClassroomCarouselToPreviousSlide(): void {
-    this.classroomCarouselIndex -= 1;
-    this.classroomCarousel.prev();
+    if (this.currentCardIndex > 0) {
+      this.currentCardIndex -= 1;
+      this.translateX += this.getCardWidth();
+      this.updateActiveDot();
+    }
   }
 
   moveClassroomCarouselToNextSlide(): void {
-    this.classroomCarouselIndex += 1;
-    this.classroomCarousel.next();
+    if (
+      this.currentCardIndex <
+      this.classroomSummaries.length - this.cardsToShow
+    ) {
+      this.currentCardIndex += 1;
+      this.translateX -= this.getCardWidth();
+      this.updateActiveDot();
+    }
   }
 
-  onClassroomNavigationIndicatorClicked(slideEvent: NgbSlideEvent): void {
-    // Extract numeric index from slide id (format: 'ngb-slide-{index}')
-    this.classroomCarouselIndex = parseInt(slideEvent?.current?.split('-')[2]);
+  moveToSlide(index: number): void {
+    this.translateX = -index * this.getCardWidth();
+    this.currentCardIndex = index;
+    this.updateActiveDot();
   }
 
-  getClassroomChunkIndices(length: number): number[] {
-    const chunkSize = 3;
-    const numChunks = Math.ceil(length / chunkSize);
-    return Array(numChunks)
-      .fill(0)
-      .map((_, index) => index);
+  getCardWidth(): number {
+    const cardContainerWidth =
+      document.querySelector('.carousel-container')?.clientWidth || 0;
+    return cardContainerWidth / this.cardsToShow;
   }
 
-  getClassroomsForChunk(
-    classroomSummaries: ClassroomSummaryDict[],
-    chunkIndex: number
-  ): ClassroomSummaryDict[] {
-    const chunkSize = 3;
-    const start = chunkIndex * chunkSize;
-    const end = start + chunkSize;
-    return classroomSummaries.slice(start, end);
+  updateActiveDot(): void {
```

Comment:
Given the same number of classroomSummaries this function produces the same results, or no? If so, do you need to call it from the functions above?

Reply:
Yes it gives the same output. the dots is just a binary array where 1 means the current slide is active and 0 mean not active.

we have this separate method because whenever we either slide of jump to any index we have to update the dots state.

and we need to call it after getting classroomSummeries to initialize it with dot[0] = 1 since we are at first slide/index.